Make Cypher 25 the default query language (#1483)
neo4j/docs-cypher: modules/ROOT/pages/queries/select-version.adoc

[[select-default-cypher-version]]
== Select the default Cypher version for a database

=== Neo4j 2026.02 or later

> Databases created on Neo4j 2026.02 have Cypher 25 as their default query language.

Author: ```suggestion
Databases created on Neo4j 2026.02 or later explicitly set Cypher 25 as their query language in the _neo4j.conf_ file: `db.query.default_language=CYPHER_25`.
```

> Existing databases keep their query language.

Author: ```suggestion
Existing databases retain  keep their _neo4j.conf_ file.
```

> Both can be adjusted via the setting link:{neo4j-docs-base-uri}/operations-manual/current/configuration/configuration-settings/#config_db.query.default_language[`db.query.default_language`].

Author: ```suggestion
Both can be adjusted via the setting link:{neo4j-docs-base-uri}/operations-manual/current/configuration/configuration-settings/#config_db.query.default_language[`db.query.default_language`].

[NOTE]
====
If the setting is missing, it defaults to `db.query.default_language=CYPHER_5`.
====
```

[[select-default-cypher-version]]
== Select the default Cypher version for a database

=== Neo4j 2026.02 or later

Databases created on Neo4j 2026.02 or later explicitly set Cypher 25 as their query language in the _neo4j.conf_ file: `db.query.default_language=CYPHER_25`.

> Existing databases retain  keep their _neo4j.conf_ file.

Author: ```suggestion
Existing databases retain their _neo4j.conf_ file.
```

> Both can be adjusted via the setting link:{neo4j-docs-base-uri}/operations-manual/current/configuration/configuration-settings/#config_db.query.default_language[`db.query.default_language`].

Author: ```suggestion
For both, a different query language version can be set with the same setting link:{neo4j-docs-base-uri}/operations-manual/current/configuration/configuration-settings/#config_db.query.default_language[`db.query.default_language`].
```

[NOTE]
====
If the setting is missing, it defaults to `db.query.default_language=CYPHER_5`.
====

In Aura, Cypher 25 is the default for newly created organisations.
The default query language attached to created databases is determined by the organisation setting "Default Cypher Version" and can be manually changed.

=== Neo4j 2025.06 - 2026.01

> Databases created on, or migrated to, Neo4j 2025.06  later will continue to have Cypher 5 as their default language (unless link:{neo4j-docs-base-uri}/operations-manual/current/configuration/configuration-settings/#config_db.query.default_language[`db.query.default_language`] is set to `CYPHER_25`).

Author: ```suggestion
Databases created on, or migrated to, Neo4j 2025.06 or later will continue to have Cypher 5 as their default language (unless link:{neo4j-docs-base-uri}/operations-manual/current/configuration/configuration-settings/#config_db.query.default_language[`db.query.default_language`] is set to `CYPHER_25`).
```
